modules/routes/variables.tf

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The ID of the project where the routes will be created"
+}
+
+variable "network_name" {
+  description = "The name of the network where routes will be created"
+}
+
+variable "network" {
+  description = "The network resource that we depend on being created first"
+  default     = null
+}
+
+variable "subnets" {
```

Comment:
We shouldn't need this.

Reply:
Its only passed into the `depends_on` block, functionality that was there before. 

Ill remove it and see what happens. I'm not entirely sure if the subnets really do need to be fully created before the routes are created.

Reply:
If it's an issue, we could possibly use an output from the subnets module to pass into the network.

Reply:
At least one of the tests fail if we remove this. 

When creating a route with a `next_hop_ip` with an ip inside a subnet it'll error out with the following when the route tries to get created before the subnet is finished. 

```
Error: Error creating Route: googleapi: Error 400: Invalid value for field 'resource.nextHopIp': '10.10.40.10'. 10.10.40.10 must lie within the address spaces of (). multi-vpc-a1-02 does not own any address space. Please create a subnetwork first., invalid
```

Reply:
Got it. In that case, how about we follow the convention we started with network peering and add an explicit `module_depends_on` variable? https://github.com/terraform-google-modules/terraform-google-network/blob/8368373300c892d3ae554f23d7cb82adbed926a8/modules/network-peering/variables.tf#L45